src/main/webapp/app/services/visualizations/scatter-plot.service.js

```diff
@@ -44,1134 +43,66 @@
-                                        return '';
-                                    }
-                                    return D3Utils.getTruncatedTick(D3Utils.shortScale(2)(d), margin.left - 8, tickLength);
-                                });
-
-                            axisLeftG.append('g')
-                                .attr('id', 'y_axis')
-                                .call(axisLeft);
-
-                            me.helper.setAxisColor(this);
-
-                            var lasso = d3.lasso()
-                                .hoverSelect(true)
-                                .closePathSelect(true)
-                                .closePathDistance(100)
-                                .items(point)
-                                .targetArea(svg);
-                            
-                            lasso.on('start', me.helper.onLassoStart(lasso, me))
-                                .on('draw', me.helper.onLassoDraw(lasso, me))
-                                .on('end', me.helper.onLassoEnd(lasso, me));
-                            
-                            svg.call(lasso);
-                        }
-
-                        if(typeof(dimensions[1]) !== 'undefined') {
-                            drawLegend.call(this, data);
-                        }
-
-                        drawPlot.call(this, data);
-                    }
-
-                    return Scatter;
-
-                })();
 
-                if(Object.keys($rootScope.updateWidget).indexOf(record.id) != -1) {
-                    if($rootScope.filterSelection.id != record.id) {
+                if (Object.keys($rootScope.updateWidget).indexOf(record.id) != -1) {
+                    if ($rootScope.filterSelection.id != record.id) {
                         var scatter = $rootScope.updateWidget[record.id];
                         scatter.updateChart(record.data);
                     }
                 } else {
-                    var scatter = new Scatter(element[0], record, getProperties(VisualizationUtils, record));
-                    scatter.renderChart();
+                    d3.select(element[0]).html('')
+                    var div = d3.select(element[0]).append('div')
+                        .attr('id', 'scatter-' + element[0].id)
+                        .style('width', element[0].clientWidth + 'px')
+                        .style('height', element[0].clientHeight + 'px')
+                        .style('overflow', 'hidden')
+                        .style('text-align', 'center')
+                        .style('position', 'relative');
+
+                    var svg = div.append('svg')
+                        .attr('width', element[0].clientWidth)
+                        .attr('height', element[0].clientHeight)
+
+                    var tooltip = div.append('div')
+                        .attr('class', 'tooltip');
+
+                    var scatter = flairVisualizations.scatter()
```

Comment:
.print(false) here. Although, it's not mandatory but just for better understanding..